pkg/controller/commonTestUtils/eventEmitterMock.go

```diff
@@ -7,18 +7,57 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sync"
 )
 
-type EventEmitterMock struct{}
+type MockEvent struct {
+	EventType string
+	Reason    string
+	Msg       string
+}
+
+type EventEmitterMock struct {
+	expectedEvents *sync.Map
+}
+
+func NewEventEmitterMock(expectedEvents ...MockEvent) *EventEmitterMock {
+	eem := &EventEmitterMock{}
+	eem.SetExpectedEvents(expectedEvents...)
+
+	return eem
+}
+
+func (eem *EventEmitterMock) SetExpectedEvents(expectedEvents ...MockEvent) {
+	events := &sync.Map{}
+	for _, event := range expectedEvents {
+		events.Store(event, false)
+	}
+
+	eem.expectedEvents = events
+}
 
 func (EventEmitterMock) Init(_ context.Context, _ manager.Manager, _ hcoutil.ClusterInfo, _ logr.Logger) {
 	/* not implemented; mock only */
 }
 
-func (EventEmitterMock) EmitEvent(_ runtime.Object, _, _, _ string) {
-	/* not implemented; mock only */
+func (eem *EventEmitterMock) EmitEvent(_ runtime.Object, eventType, reason, msg string) {
+	eem.expectedEvents.Store(MockEvent{
+		EventType: eventType,
+		Reason:    reason,
+		Msg:       msg,
+	}, true)
 }
 
 func (EventEmitterMock) UpdateClient(_ context.Context, _ client.Reader, _ logr.Logger) {
 	/* not implemented; mock only */
 }
+
+func (eem EventEmitterMock) GotExpectedEvent() bool {
```

Comment:
---This comment is not critical. Just an improvement idea. Current form is already OK. ---

- Instead of `expectedEvents` map, we can initialize it with an empty map named `eventStore`.
- Instead of setting map value in `EmitEvent`, we can insert the event into the `eventStore`
- Instead of `GotExpectedEvent`, we can implement two methods
   - ContainEvents (events []MockEvent) bool -> Checks the events exist in the `eventStore` or not. It can be useful when you want to check specific events but you don't care other emitted events in your test. It can be useful for negative checks as well like you don't expect having a specific event in the emitter.
   - EqualsTo(events []MockEvent) -> Does  equality check. More events in the emitter is not accepted. It can be useful when you want to check all events. No more no less.

Reply:
Done